command.c

```diff
@@ -5,33 +5,33 @@
 
 #include "nsq.h"
 
-static const int MAX_BUF_SIZE = 128;
-static const int MIN_BUF_SIZE = 64;
-static const int BUF_DELTER   = 2;
+static const int defaultBufSize = 32;
+static const int restBufSize    = 16;
+static const int bufDelter      = 1;
 
-void *nsq_buf_malloc(size_t buf_size, size_t n, size_t l)
+void *nsq_buf_malloc(size_t *buf_size, size_t n, size_t l)
 {
-    if (buf_size - n >= MIN_BUF_SIZE || buf_size - l >= MIN_BUF_SIZE) {
+    if ((int)(*buf_size - n) >= restBufSize || (int)(*buf_size - l) >= restBufSize) {
```

Comment:
Can you put `nsq_buf_malloc()` and `nsq_buffer_add()` fixes into a separate pull request please?

Also, I'm still not sure this is right. Imagine `*buf_size == 64`, `n == 40`, and `l == 40`. In this case, this returns NULL  to mean no resize needed, but it needs to double the buffer size for `l` more bytes to fit after the existing `n` used bytes.

Reply:
Yes, it's definately not ok. All right, let me fix next time.

Reply:
... I would like to review and merge the `nsq_buffer_add()` changes first, and then other parts of this PR also in separate pull requests after that.

(I'm not a fan of git submodules ... in this case it's very small dependency, so  I had thought of just copying the sources into this repo, but I also have another idea to have the Makefile fetch the files only if they are out-of-date, I'll make a pull-request with that.)

Reply:
I just thought that there was no need to install libevbuffsock as a standard lib, as the connection socket was not so perfectly packed or no need to be packed into the libevbuffsock in my shallow opinion.

Reply:
agreed, I just don't like  git submodules :)